### ISSUE TITLE
[BUG] HeaderCollection attribute doesn't seem to work with POSTs

### ISSUE THREAD

**Issue by uhfath** [reactiveui/refit#1102]
As described in docs I use [HeaderCollection] attribute to supply some dynamic headers to my requests:

```C#
[Get("/{bucket}")]
Task<ListBucketResult> ListObjectsAsync(string bucket, ListObjectsRequest request, [HeaderCollection]IDictionary<string, string> headers, CancellationToken cancellationToken = default);

[Post("/{bucket}/{key}?uploads")]
Task<InitiateMultipartUploadResult> CreateMultipartUploadAsync(string bucket, string key, [HeaderCollection]IDictionary<string, string> headers, CancellationToken cancellationToken = default);
```
While it works perfectly with GETs it just doesn't work the same way with POSTs:

```c#
One or more errors occurred. (The type System.Collections.Generic.Dictionary`2[[System.String, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.String, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]] is not supported because it implements IDictionary.)
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at TestMinioClient.Program.Main(String[] args) in D:\Projects\TestMinioClient\Program.cs:line 237
```

Also with GETs everything is working when supplied header collection is null. But an exception (value is null) is thrown when using the same with POSTs.

Tested using 6.0.24

**Comment by james-s-tayler** [comment]
Interesting. We've got a test where `[HeaderCollection]` is used in conjunction with `POST` and that works fine, though the difference I can see here is that test includes a `[Body]` in the method signature. Looking at the code sample and the stacktrace my initial suspicions are that 1) it's trying to serialize the `[HeaderCollection]` dictionary for some reason, and 2) your method signature is causing it to do so for some reason. Gut feel is a bug triggered by not having a `[Body]`. Will investigate.

**Comment by james-s-tayler** [comment]
@uhfath I have opened a PR to fix this.

**Comment by uhfath** [comment]
@james-s-tayler thanks for taking a look into this and fixing the issue. Will test as soon as I get the chance.

**Comment by james-s-tayler** [comment]
@uhfath no worries. Looks like the fix will be released in `6.0.48` so you'll have to wait until that's available in NuGet https://www.nuget.org/packages/Refit/

**Comment by ChrisPulman** [comment]
Issue notified as resolved.
